src/hooks/useClickAway.ts

```diff
@@ -0,0 +1,34 @@
+import type { RefObject } from 'react';
+import { useEffect, useRef } from 'react';
+
+const useClickAway = <T extends HTMLElement>(
+  handler: (e?: Event) => void
+): RefObject<T> => {
+  const ref = useRef<T>(null);
+  const saveHandler = useRef(handler);
+
+  useEffect(() => {
+    saveHandler.current = handler;
+  }, [handler]);
+
+  useEffect(() => {
+    const el = ref.current;
+    if (!el) return;
+
+    const handleEvent: EventListener = (e) => {
+      if (!el.contains(e.target as Node)) {
```

Comment:
```suggestion
      if (!el.contains(e.target as HTMLElement)) {
```
HTMLElement 는 Element 인터페이스를 extends하고, Element 는 Node 인터페이스를 extends한다고 하는데, 혹시  Node 로 as 를 주신 특별한 이유가 있을까요?